app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestDetailViewFragment.java

```diff
@@ -0,0 +1,167 @@
+//        this.btn2.setText(getString(R.string.deny));
+//
+//        this.btn1.setOnClickListener(this::btnAccept);
+//        this.btn2.setOnClickListener(this::btnDeny);
+//
+//        this.btn1.setVisibility(View.VISIBLE);
+//        this.btn2.setVisibility(View.VISIBLE);
+
+        // case accept
+//        this.btn1.setText(getString(R.string.set_location_label));
+//        this.btn2.setText(getString(R.string.lend_book));
+//
+//        this.btn1.setOnClickListener(this::btnSetLocation);
+//        this.btn2.setOnClickListener(this::btnLendBook);
+//
+//        //TODO: get pickup location from book
+//        this.bookDetailView.findViewById(R.id.request_exchange).setVisibility(View.VISIBLE);
+//
+//        this.btn1.setVisibility(View.VISIBLE);
+//        this.btn2.setVisibility(View.VISIBLE);
+
+        // case bPending
+//        this.btn1.setText(getString(R.string.wait_borrower));
+//
+//        this.btn1.setVisibility(View.VISIBLE);
+
+        // case rPending
+//        this.btn1.setText(getString(R.string.accept_return));
+//        this.btn2.setText(getString(R.string.view_location));
+//
+//        this.btn1.setOnClickListener(this::btnAcceptReturn);
+//        this.btn2.setOnClickListener(this::btnViewLocation);
+//
+//        this.btn1.setVisibility(View.VISIBLE);
+//        this.btn2.setVisibility(View.VISIBLE);
+
+        return bookDetailView;
+    }
+
+    private void btnViewLocation(View view)
+    {
+        //TODO: actually do the stuff
+        // launch ViewLocation
+    }
+
+    private void btnAcceptReturn(View view)
+    {
+        //TODO: actually do the stuff
+        // Launch Scan ISBN
+        // update book status
+        // launch RequestsFragment
+    }
+
+    private void btnLendBook(View view)
+    {
+        //TODO: actually do the stuff
+        // Launch Scan ISBN
+        // update book status
+        // launch RequestsFragment
+    }
+
+    private void btnSetLocation(View view)
+    {
+        //TODO: actually do the stuff
+        // launch SetLocation
+    }
+
+    private void btnDeny(View view)
+    {
+        //TODO: actually do the stuff
+        // get borrower id
+        // remove borrower from book request list
+        // launch RequestsFragment
+    }
+
+    private void btnAccept(View view)
+    {
+        //TODO: actually do the stuff
+        // change book status to accepted (clear book request list)
+        // launch new RequestDetailViewFragment with updated book
+    }
+
+    /**
+     * Update the textfields for book detail view based on the given book
+     *
+     * @param book The book containing the data to populate the textfields with
+     */
+    public void updateView(Book book)
+    {
+        // Set the content based on the book that was selected
+        super.updateView(book);
+
+        TextView status = bookDetailView.findViewById(R.id.book_details_status);
+        status.setText(book.getStatus().toString() + " " + getString(R.string.detail_join));
```

Comment:
I'm having trouble understanding this line, does this hardcode "borrowed by"?

Reply:
This makes the book status descriptor the book status + by (borrowed by, accepted by, etc)
It will change after the statuses are pushed and I can properly determine what to put there.

Reply:
ok